lib/matplotlib/axes/_base.py

```diff
@@ -2871,8 +2871,12 @@ def set_xlim(self, left=None, right=None, emit=True, auto=False, **kw):
 
         self._process_unit_info(xdata=(left, right))
         if left is not None:
+            if isinstance(left,float) and (math.isnan(left) or np.isinf(left)):
```

Comment:
I think it still works, but I think we should leverage as much as possible numpy considering it is one of our dependencies.